File: openoversea/src/main/java/com/bytedance/sdk/open/tiktok/share/ShareDataHandler.java
package com.bytedance.sdk.open.tiktok.share;

import android.os.Bundle;

import androidx.annotation.Nullable;

import com.bytedance.sdk.open.tiktok.common.constants.Constants;
import com.bytedance.sdk.open.tiktok.common.handler.IApiEventHandler;
import com.bytedance.sdk.open.tiktok.common.handler.IDataHandler;

/**
 * sharing data parser
 */
public class ShareDataHandler implements IDataHandler {
    @Override
    public boolean handle(int type, @Nullable Bundle bundle, @Nullable IApiEventHandler eventHandler) {
        if (bundle == null || eventHandler == null) {
            return false;
        }
        if (type == Constants.TIKTOK.SHARE_REQUEST) {
            Share.Request request = new Share.Request();
            request.fromBundle(bundle);
            if (request.validate()) {
                eventHandler.onReq(request);
                return true;
            } else {
                return false;
            }
        } else if (type == Constants.TIKTOK.SHARE_RESPONSE) {
            Share.Response response = new Share.Response();
            response.fromBundle(bundle);
            if (response.validate()) {
                eventHandler.onResp(response);
                return true;
            } else {
                return false;
            }
        }
        return false;
    }
}

File: openoversea/src/main/java/com/bytedance/sdk/open/tiktok/share/ShareDataHandler.kt
package com.bytedance.sdk.open.tiktok.share

import android.os.Bundle
import com.bytedance.sdk.open.tiktok.common.constants.Constants
import com.bytedance.sdk.open.tiktok.common.handler.IApiEventHandler
import com.bytedance.sdk.open.tiktok.common.handler.IDataHandler

class ShareDataHandler : IDataHandler {
    override fun handle(type: Int, bundle: Bundle, eventHandler: IApiEventHandler): Boolean {
        return when (type) {
            Constants.TIKTOK.SHARE_REQUEST -> {
                val request = Share.Request()
                request.fromBundle(bundle)
                if (request.validate()) {
                    eventHandler.onRequest(request)
                    true
                } else {
                    false
                }
            }
            Constants.TIKTOK.SHARE_RESPONSE -> {
                val response = Share.Response()
                response.fromBundle(bundle)
                if (response.validate()) {
                    eventHandler.onResponse(response)
                    true
                } else {
                    false
                }
            }
            else -> false
        }
    }
}
